src/server/game/AI/SmartScripts/SmartScript.h

```diff
@@ -122,6 +123,7 @@ class TC_GAME_API SmartScript
         GameObject* go;
         ObjectGuid goOrigGUID;
         AreaTriggerEntry const* trigger;
+        AreaTriggerTemplate const* areaTriggerTemplate;
```

Comment:
Don't support SAI for now, that would require `class AreaTrigger` to support SAI in the first place (you would need two new SMART_SCRIPT_TYPE_)

Reply:
but it does?
That is, I tested SAI and it works with the server side impelmentation as I currently did it?

Reply:
It worked but you had id conflict with old db2 based areatriggers

Reply:
> It worked but you had id conflict with old db2 based areatriggers
What do you mean exactly by that?
I mean yes the old area trigger would still trigger, but I'm pretty sure they had no effectss.
I use a new source type for the SmartAI, and only the new area trigger i've created in the db (id=2, IsServer=1) has "SmartTriggerServer" on it's script name.

Reply:
I took a 2nd look and now I understand what you're saying.

I implemented all the thing that are required for SAI, but removed them in one of the commits. You can see them here:
https://github.com/TrinityCore/TrinityCore/pull/25971/commits/e7764cd341788ee4746b9ebe32a7f049c734833b

I can bring it back it shouldn't be much problem, but if you prefer we can leave it for another issue.

Reply:
Up to you if you want to support it fully or not at all, just don't leave this partially broken code

Reply:
I'm gonna leave tihs out of this PR and I've created a new ticket to add SAI to this when it's done